streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamTransformValuesTest.java

```diff
@@ -94,14 +97,82 @@ public void close() {
 
         assertEquals(4, processor.processed.size());
 
-        driver.punctuate(2);
-        driver.punctuate(3);
-
-        String[] expected = {"1:10", "10:110", "100:1110", "1000:11110", "null:2", "null:3"};
+        String[] expected = {"1:10", "10:110", "100:1110", "1000:11110"};
 
         for (int i = 0; i < expected.length; i++) {
             assertEquals(expected[i], processor.processed.get(i));
         }
     }
 
+    @Test
+    public void shouldNotAllowValueTransformerToCallInternalProcessorContextMethods() {
```

Comment:
I stand by what i said, but I'll leave it up to you. It isn't a deal breaker for me!